pkg/controller/restore_controller.go

```diff
@@ -462,43 +468,92 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return nil, errors.WithStack(err)
+	location, err := c.backupLocationLister.BackupStorageLocations(c.namespace).Get(info.backup.Spec.StorageLocation)
+	if err != nil {
+		return backupInfo{}, errors.WithStack(err)
 	}
 
-	logContext := c.logger.WithField("backupName", name)
+	info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+	if err != nil {
+		return backupInfo{}, errors.Wrap(err, "error initializing object store")
+	}
+	info.bucketName = location.Spec.ObjectStorage.Bucket
+
+	return info, nil
+}
 
-	logContext.Debug("Backup not found in backupLister, checking object storage directly")
-	backup, err = c.getBackup(objectStore, c.bucket, name)
+// fetchFromBackupStorage checks each backup storage location, starting with the default,
+// looking for a backup that matches the given backup name.
+func (c *restoreController) fetchFromBackupStorage(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	locations, err := c.backupLocationLister.BackupStorageLocations(c.namespace).List(labels.Everything())
 	if err != nil {
-		return nil, err
+		return backupInfo{}, err
+	}
+
+	idx := 1
+	orderedLocations := make([]*api.BackupStorageLocation, len(locations))
+	for i := 0; i < len(locations); i++ {
+		if locations[i].Name == c.defaultBackupLocation {
+			orderedLocations[0] = locations[i]
+		} else {
+			orderedLocations[idx] = locations[i]
+			idx++
+		}
 	}
 
-	// ResourceVersion needs to be cleared in order to create the object in the API
-	backup.ResourceVersion = ""
-	// Clear out the namespace too, just in case
-	backup.Namespace = ""
+	logContext := c.logger.WithField("backupName", backupName)
+	var info backupInfo
+	for _, location := range orderedLocations {
+		info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+		if err != nil {
+			logContext.WithError(errors.WithStack(err)).Error("Unable to get an object store in the location ", location.Name)
+			continue
+		}
 
-	created, createErr := c.backupClient.Backups(c.namespace).Create(backup)
-	if createErr != nil {
-		logContext.WithError(errors.WithStack(createErr)).Error("Unable to create API object for Backup")
-	} else {
-		backup = created
+		info.backup, err = c.getBackup(info.objectStore, location.Spec.ObjectStorage.Bucket, backupName)
+		if err != nil {
+			logContext.WithError(errors.WithStack(err)).Error("Unable to find a backup in the location ", location.Name)
+			continue
+		}
+
+		// ResourceVersion needs to be cleared in order to create the object in the API
+		info.backup.ResourceVersion = ""
+		// Clear out the namespace, in case the backup was made in a different cluster, with a different namespace
+		info.backup.Namespace = ""
+
+		info.backup, err = c.backupClient.Backups(c.namespace).Create(info.backup)
+		if err != nil {
+			logContext.WithError(errors.WithStack(err)).Error("Unable to create API object for Backup")
```

Comment:
I think we should just return the error here -- we shouldn't be continuing the loop in this scenario

Reply:
Don't we want to move on to the next location?

Reply:
Don't think so - we've already found the backup we're looking for.  Continuing to look elsewhere wouldn't help

Reply:
Ok I see. I didn't make my reasoning at all clear. Here it is: I thought a valid backup to use was one that was found + created. So this is not the case? If we create it or not doesn't matter?

Reply:
So in general there *should* only be a single backup with a given name.  In which case, once we've found it, continuing to search elsewhere if we can't create it is pointless, because we've already found the one we're looking for.

In the current state, there **could** be >1 backup with a given name if the user is importing backups from other clusters.  I think you're arguing that in this case, if we find one, but the API call to sync it into the cluster fails, we should just try another one?  This is already kind of an unsupported scenario so it's a little hard to reason about, but I'd say that if creating the first one failed, then attempting to create any other one with the same name will probably fail too (maybe the API's down, maybe something's wrong with RBAC, maybe a backup with the same name got created in between when we checked for existence and when we went to sync one in, ...).  And, soon, the possibility of this scenario existing will go away.

Reply:
Ah. I really appreciate the explanation! Thank you.